records_mover/airflow/hooks/records_hook.py

```diff
@@ -50,12 +51,15 @@ def _url_resolver(self) -> UrlResolver:
                            gcs_client_getter=lambda: None,
                            gcp_credentials_getter=lambda: None)
 
-    def _db_driver(self, db: Union[sqlalchemy.engine.Engine,
-                                   sqlalchemy.engine.Connection]) -> DBDriver:
+    def _db_driver(self, db: Optional[Union[sqlalchemy.engine.Engine,
```

Comment:
is this supposed to have `@property` like `_url_resolver` above does?

Reply:
I don't think so, strangely enough _db_driver seems to be a function and not a property. I'm going to check that I'm right on this.

Reply:
I think I am, this is very awkward syntax but: it seems like it's not actually a property of the class but that this is instantiating a db_driver through the db_driver factory so I don't think this would be appropriate to assign to the "getter" method of the @property _db_driver. It's possible that it could be assigned to the setter method for the property _db_driver property but that might have downstream effects that I don't want.

I agree that this wasn't really the best method name (should have been create_db_driver or something like that) but I'm hesitant to change it at this point because it gets called all over the place.

Reply:
Let's talk through this at our meeting Monday just to make sure that I'm not lost in my thinking.

Reply:
Resolved per our chat